doc/source/user_guide/text.rst

```diff
@@ -94,6 +94,10 @@ l. For ``StringDtype``, :ref:`string accessor methods<api.series.str>`
 2. Some string methods, like :meth:`Series.str.decode` are not available
    on ``StringArray`` because ``StringArray`` only holds strings, not
    bytes.
+3. In comparision operations, :class:`StringArray` and ``Series`` backed
+   by a ``StringArray`` will return an object with :class:`BooleanDtype`,
+   rather than a ``bool`` or ``object`` dtype object, depending on whether
```

Comment:
Currently, I don't think you get object dtype (None or NaN compares to False, so it still is a boolean result)